cflib/crazyflie/__init__.py

```diff
@@ -153,6 +155,11 @@ def __init__(self, link=None, ro_cache=None, rw_cache=None):
         self.fully_connected.add_callback(
             lambda uri: logger.info('Callback->Connection completed [%s]', uri))
 
+        self.connected.add_callback(
+            lambda uri: self.link_statistics.start())
+        self.disconnected.add_callback(
+            lambda uri: self.link_statistics.stop())
```

Comment:
When using this library with the cfclient, upon reconnect with the cfclient still active, I get an error:

```
INFO:cflib.crazyflie:Callback->Connection setup finished [radio://0/40/2M/E7E7E7E705]
ERROR:cflib.crazyflie:Exception while doing callback on port [2]

Traceback (most recent call last):
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\__init__.py", line 441, in run
    cb.callback(pk)
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\param.py", line 548, in _new_packet_cb
    self._done_callback()
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\__init__.py", line 180, in _param_toc_updated_cb
    self.connected.call(self.link_uri)
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\utils\callbacks.py", line 54, in call
    cb(*args)
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\__init__.py", line 159, in <lambda>
    lambda uri: self.link_statistics.start())
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\link_statistics.py", line 68, in start
    self.latency.start()
  File "C:\Users\kimbe\Development\bitcraze\python\crazyflie-lib-python\cflib\crazyflie\link_statistics.py", line 107, in start
    self._ping_thread_instance.start()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\threading.py", line 930, in start
    raise RuntimeError("threads can only be started once")
RuntimeError: threads can only be started once
```

But the .stop() does seem to close the thread, or at least it is intended to do that. So perhaps you can see if that actually happens.

Reply:
Good find! The threads were properly closed, but Python threading does not allow to start the same thread instance after stopping. Fixed with a6a92155d2b904f9a1e858368781dd6683cda7db.